public/components/palantir/index.tsx

```diff
@@ -0,0 +1,197 @@
+      iconType={logos.Chat.url}
+      iconSide="right"
+      onClick={onAnchorClick}
+    >
+      <span className="palantirAnchorContent">{children}</span>
+    </EuiButtonEmpty>
+  );
+
+  // TODO: i18n && delay
+  // TODO: compressed loading component (empty prompt)
+  // TODO: --- loading 2-3 seconds
+  const getPalantirInputByChildKey = () => {
+    let palantir;
+    Children.forEach(children, (child) => {
+      if (isValidElement(child) && child.key && registry.get(child.key as string)) {
+        palantir = registry.get(child.key as string);
+      }
+    });
+    if (!palantir) throw Error('Child key not found in registry.');
+    return palantir;
+  };
+
+  const palantirInput = input ?? getPalantirInputByChildKey();
+
+  const suggestionsPopoverFooter: React.ReactNode = (
+    <EuiPopoverFooter>
+      <EuiText size="xs" color="subdued">
+        Available suggestions
+      </EuiText>
+      <EuiBadge
+        color="hollow"
+        iconType="chatRight"
+        iconSide="left"
+        onClick={() => onSubmitClick(palantirInput)}
+        onClickAriaLabel="Click for suggestion"
+      >
+        {palantirInput.suggestion}
+      </EuiBadge>
+    </EuiPopoverFooter>
+  );
+
+  const generatePopoverBody: React.ReactNode = (
+    <EuiButton onClick={() => toasts.addDanger('To be implemented...')}>Generate summary</EuiButton>
+  );
+
+  const summaryPopoverBody: React.ReactNode = (
+    <EuiText size="s">
+      <>{palantirInput.summary}</>
+    </EuiText>
+  );
+
+  const summaryWithSuggestionsPopoverBody: React.ReactNode = (
+    <>
+      {summaryPopoverBody}
+      {suggestionsPopoverFooter}
+    </>
+  );
+
+  const [chatValue, setChatValue] = useState('');
+  const chatPopoverBody: React.ReactNode = (
+    <EuiFlexGroup>
+      <EuiFlexItem grow={6}>
+        <EuiFormRow>
+          <EuiFieldText
+            placeholder="Ask a question"
+            value={chatValue}
+            onChange={(e) => setChatValue(e.target.value)}
+          />
+        </EuiFormRow>
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiButton
+          fill
+          iconType="returnKey"
+          iconSide="right"
+          onClick={() => toasts.addDanger(`To be implemented... received ${chatValue}`)}
+        >
+          Go
+        </EuiButton>
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  );
+
+  const chatWithSuggestionsPopoverBody: React.ReactNode = (
+    <>
+      {chatPopoverBody}
+      {suggestionsPopoverFooter}
+    </>
+  );
+
+  const getPopoverBody = () => {
+    switch (palantirInput.type) {
+      case 'suggestions':
+        return suggestionsPopoverFooter;
+      case 'generate':
+        return generatePopoverBody;
+      case 'summary':
+        return summaryPopoverBody;
+      case 'summaryWithSuggestions':
+        return summaryWithSuggestionsPopoverBody;
+      case 'chat':
+        return chatPopoverBody;
+      case 'chatWithSuggestions':
+        return chatWithSuggestionsPopoverBody;
```

Comment:
<img width="349" alt="Screenshot 2024-01-02 at 4 04 14 PM" src="https://github.com/opensearch-project/dashboards-assistant/assets/12060704/18da888f-1651-4b22-b93e-b0eebc0edb14">

Reply:
For these, can we use the compressed version of form elements?